### ISSUE TITLE
IR camera works on Android 11, doesn't work on Android 12 or later

### ISSUE THREAD

**Issue by ssorgatem** [phhusson/ulefone-armor-9-gsi#4]
On Android 12 or 13 GSIs, the MyFLIR app appears to work, except it doesn't get an image from the camera.

I can hear the camera clicking though.

I even tried using the MyFLIR apk from the Power Armor T18, which runs Android12, with no luck (same behaviour).
The Armor T18 seems to be using a different set of libraries and files too...

**Comment by phhusson** [comment]
I seem to remember it worked for me, with maybe the additional
requirement of needing to run selinux permissive (`setenforce 0`)
I'll try checking on mine this week.

**Comment by ssorgatem** [comment]
I just tried with a LineageOS 18.1 GSI and I have the same issue, even though it's Android 11 too. May be related to  Selinux differences?

I'll have to look into how to mess with it.

**Comment by ssorgatem** [comment]
Somehow I flashed the very same image where I had it working (latest system-roar-arm64-ab-gapps.img) and now it's not working there either... I'm at a loss.
How can I debug it?

**Comment by ssorgatem** [comment]
Yes! "sentenforce 0" did the trick and I now have it working on LineageOS 19.1 (Android 12).

Is that always needed or can this be fixed somehow in the generate.sh script?

**Comment by phhusson** [comment]
It could (and should because selinux permissive really removes all Android
security mechanisms) probably be fixed, but it needs a bit of work.

**Comment by ssorgatem** [comment]
Well, I decided to try Android 13 (Pixel Experience Plus GSI) and I'm running into the very same issue I mentioned initially, even with permissive SELinux.
